### ISSUE TITLE
brew install cups :  error: unknown type name 'gss_OID'

### ISSUE THREAD

**Issue by tseemann** [Homebrew/linuxbrew-core#19707]
- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

install cups  # actually install a java app, but jdk needs cups?

## What happened (include command output)

```
==> Downloading https://github.com/apple/cups/releases/download/v2.3.1/cups-2.3.1-source.tar.gz
Already downloaded: /home/linuxbrew/.cache/Homebrew/downloads/ef5011ed17aedae36daddee667ae8b670abe20e895780d11d57f40b4bed3c0e9--cups-2.3.1-source.tar.gz
==> ./configure --with-components=core --without-bundledir --prefix=/home/linuxbrew/.linuxbrew/Cellar/cups/2.3.1
==> make install
Last 15 lines from /home/linuxbrew/.cache/Homebrew/Logs/cups/02.make:
make[1]: *** [../Makedefs:270: dest.o] Error 1
In file included from cups-private.h:21:0,
                 from dest-job.c:13:
http-private.h:263:3: error: unknown type name 'gss_OID'
   gss_OID   gssmech; /* Authentication mechanism */
   ^
http-private.h:264:3: error: unknown type name 'gss_ctx_id_t'
   gss_ctx_id_t  gssctx;  /* Authentication context */
   ^
http-private.h:265:3: error: unknown type name 'gss_name_t'
   gss_name_t  gssname; /* Authentication server name */
   ^
make[1]: *** [../Makedefs:270: dest-job.o] Error 1
make[1]: Leaving directory '/tmp/cups-20200225-5715-1jzt133/cups-2.3.1/cups'
make: *** [Makefile:167: install-data] Error 2
```

## What you expected to happen

install cups

## Step-by-step reproduction instructions (by running `brew install` commands)

brew install cups

**Comment by issyl0** [comment]
Can we have `brew config` and `brew doctor` output as per the issue template please?

**Comment by tseemann** [comment]
```
HOMEBREW_VERSION: 2.2.6-37-g670227b
ORIGIN: https://github.com/Homebrew/brew
HEAD: 670227bdc83348cecd7bd8ffc039e2a851f7470f
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: a20203b1462eaea4af92a7b2f2247d7ea0bff381
Core tap last commit: 5 hours ago
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CACHE: /home/linuxbrew/.cache/Homebrew
HOMEBREW_LOGS: /home/linuxbrew/.cache/Homebrew/Logs
HOMEBREW_BINTRAY_KEY: set
HOMEBREW_BINTRAY_USER: tseemann
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_GITHUB_USER: tseemann
HOMEBREW_NO_AUTO_UPDATE: 1
HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK: 1
HOMEBREW_NO_INSTALL_CLEANUP: 1
HOMEBREW_SYSTEM_CURL_TOO_OLD: 1
HOMEBREW_VISUAL: joe
CPU: hexa-core 64-bit haswell
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby
Clang: N/A
Git: 2.25.0 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 7.68.0 => /home/linuxbrew/.linuxbrew/opt/curl/bin/curl
Kernel: Linux 3.10.0-1062.9.1.el7.x86_64 x86_64 GNU/Linux
OS: Red Hat Enterprise Linux Server release 7.7 (Maipo) (Maipo)
Host glibc: 2.17
/usr/bin/gcc: 4.8.5
glibc: 2.23
gcc: 5.5.0_4
xorg: 20170115_1
```

**Comment by tseemann** [comment]
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /bin/pkg-config
  /bin/rarian-sk-config
  /bin/scrollkeeper-config
  /bin/freetype-config
  /bin/libpng-config
  /bin/brltty-config
  /bin/firewall-config
  /bin/libpng15-config
  /bin/gdlib-config
  /bin/python2-config
  /bin/python2.7-config
  /bin/nspr-config
  /bin/nss-softokn-config
  /bin/nss-config
  /bin/nss-util-config
  /bin/apr-1-config
  /bin/pcre-config
  /bin/apu-1-config
  /bin/gpg-error-config
  /bin/libgcrypt-config
  /bin/xml2-config
  /bin/python-config
  /bin/icu-config
  /bin/ncurses5-config
  /bin/ncursesw5-config
  /bin/krb5-config
  /bin/libgnutls-config
  /bin/curl-config
  /bin/kde4-config
  /bin/sane-config
  /bin/net-snmp-config
  /bin/cups-config
  /bin/xslt-config
  /bin/sdl-config
  /bin/pcap-config

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  numpy
  scipy
  tcl-tk
  roary

Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install cups

Run `brew missing` for more details.

**Comment by tseemann** [comment]
```
2020-02-25 09:16:40 +1100

make
install

Making all in cups...
make[1]: warning: -j6 forced in submake: resetting jobserver mode.
make[1]: Entering directory '/tmp/cups-20200225-11391-q56diq/cups-2.3.1/cups'
Compiling raster-interstub.c...
Compiling raster-stubs.c...
Compiling array.c...
Compiling auth.c...
Compiling debug.c...
Compiling dest.c...
In file included from cups-private.h:21:0,
                 from auth.c:17:
http-private.h:263:3: error: unknown type name 'gss_OID'
   gss_OID   gssmech; /* Authentication mechanism */
   ^
http-private.h:264:3: error: unknown type name 'gss_ctx_id_t'
   gss_ctx_id_t  gssctx;  /* Authentication context */
   ^
http-private.h:265:3: error: unknown type name 'gss_name_t'
   gss_name_t  gssname; /* Authentication server name */
   ^
In file included from cups-private.h:21:0,
                 from debug.c:14:
http-private.h:263:3: error: unknown type name 'gss_OID'
   gss_OID   gssmech; /* Authentication mechanism */
   ^
http-private.h:264:3: error: unknown type name 'gss_ctx_id_t'
   gss_ctx_id_t  gssctx;  /* Authentication context */
```

**Comment by iMichka** [comment]
`jdk` needs cups, and I find no way to disable that ... looks like you can print with java :)

**Comment by issyl0** [comment]
I found https://stackoverflow.com/a/19584147 - if you add `--disable-gssapi` to the `./configure` flags for `cups`, does that get you further?

**Comment by tseemann** [comment]
@issyl0 yes that fixed it! i made a PR - is that right?

**Comment by greghogan** [comment]
@issyl0 11d3c64fdd56ffbe97c077d40d0a7e6a9dac9cb9 is breaking, the flag is "--disable-gssapi" not "--without-gssapi".

[cups-2.3.1] $ ./configure --help | grep gssapi
  --disable-gssapi        disable GSSAPI support

**Comment by issyl0** [comment]
@greghogan Oops - I even wrote that down the first time! :flushed: I'm not at a computer right now to fix and test it - would you be willing to submit a PR?

**Comment by issyl0** [comment]
Fixed in #19816 which is going through CI checks now. Thanks for the report and for using Homebrew on Linux, @greghogan!
